v1/ao/internal/reporter/reporter_grpc.go

```diff
@@ -1102,7 +1107,12 @@ func (c *grpcConnection) InvokeRPC(exit chan struct{}, m Method) error {
 				v := fmt.Sprintf("%d|%d", m.RequestSize(), c.maxReqBytes)
 				err = errors.Wrap(errRequestTooBig, v)
 			} else {
-				err = m.Call(ctx, c.client)
+				if m.ServiceKey() != "" {
+					err = m.Call(ctx, c.client)
+				} else {
+					err = nil
```

Comment:
In this case an actual gRPC call will be made with an empty string? (hence trigger the error?)

Would it be more preferable to simply not make a call? I have some concerns if actual calls are made:
1. Some extra overhead which otherwise can be avoided
2. It is probably nicer if we avoid making calls that trigger errors on collector side? It might trigger some extra logs or handling on that end

But again, I don't know the code too well. It's just from my "java" mind that it's better to avoid errors if we know that it will trigger errors. It's just my "java" mind though, which might not apply in go-lang world 😅

Reply:
No. If the service key is empty the RPC call won't be invoked.

Reply:
how did I miss `m.ServiceKey() != ""` ... 😓